api/src/main/java/org/ccci/idm/user/User.java

```diff
@@ -411,12 +412,15 @@ public void setCruProxyAddresses(Collection<String> cruProxyAddresses) {
         this.cruProxyAddresses = cruProxyAddresses;
     }
 
-    public Collection<String> getCruPasswordHistory() {
-        return cruPasswordHistory;
+    public Set<String> getCruPasswordHistory() {
+        return ImmutableSet.copyOf(cruPasswordHistory);
     }
 
-    public void setCruPasswordHistory(final Collection<String> cruPasswordHistory) {
-        this.cruPasswordHistory = cruPasswordHistory;
+    public void setCruPasswordHistory(@Nullable final Collection<String> history) {
+        cruPasswordHistory.clear();
+        if (history != null) {
+            cruPasswordHistory.addAll(history);
```

Comment:
A non-serializable (by definition) `ArrayList$SubList` was being set as the `cruPasswordHistory` in a `User`. 

It's the `User` that needs to be serialized?